test/FunctionalTests/Microsoft.OData.Core.Tests/Json/ODataJsonWriterUtilsTests.cs

```diff
@@ -86,10 +89,43 @@ public void WriteError_WritesTargetAndDetails()
                 error,
                 includeDebugInformation: false,
                 maxInnerErrorDepth: 0,
-                writingJsonLight: false);
+                writingJsonLight: false,
+                skipNullProperties:false);
             var result = stringWriter.GetStringBuilder().ToString();
             result.Should().Be(@"{""error"":{""code"":"""",""message"":"""",""target"":""any target""," +
                 @"""details"":[{""code"":""500"",""target"":""any target"",""message"":""any msg""}]}}");
         }
+
+
+        [Fact]
+        public void WriteError_InnerErrorWithDetails()
+        {
+            IDictionary<string, ODataValue> properties = new Dictionary<string, ODataValue>();
+            properties.Add("stacktrace", "NormalString".ToODataValue());
+            properties.Add("MyNewObject", new ODataResourceValue(){TypeName = "ComplexValue", Properties = new List<ODataProperty>(){ new ODataProperty(){ Name = "NestedResourcePropertyName", Value = "NestedPropertyValue"}}});
+            var error = new ODataError
+            {
+                Target = "any target",
+                Details =
+                    new[] { new ODataErrorDetail { ErrorCode = "500", Target = "any target", Message = "any msg" } },
+                InnerError = new ODataInnerError(properties, "innererror", new ODataInnerError(properties, "nested", null))
+            };
+
+            ODataJsonWriterUtils.WriteError(
+                jsonWriter,
+                enumerable => { },
+                error,
+                includeDebugInformation: true,
+                maxInnerErrorDepth: 5,
+                writingJsonLight: false,
+                skipNullProperties: false);
+            var result = stringWriter.GetStringBuilder().ToString();
+            result.Should().Be("{\"error\":{\"code\":\"\",\"message\":\"\",\"target\":\"any target\",\"details\":[{\"code\":\"500\",\"target\":\"any target\",\"message\":\"any msg\"}],\"innererror\":{\"stacktrace\":\"NormalString\",\"MyNewObject\":{\"NestedResourcePropertyName\":\"NestedPropertyValue\"},\"message\":\"\",\"type\":\"\",\"innererror\":{\"stacktrace\":\"NormalString\",\"MyNewObject\":{\"NestedResourcePropertyName\":\"NestedPropertyValue\"},\"message\":\"\",\"type\":\"\"}}}}");

```

Comment:
>"message\":\"\",\"type\" [](start = 306, length = 24)

If "message" or "type" is empty string or null string, should they be in the payload?